### DISCUSSION TITLE
XMLHttpRequest error while using flutter web and http.Request

### DISCUSSION THREAD

**Discussion by AvoMandjian** [serverpod/serverpod#877] [Q&A]
I have the project deployed on GCP with cloud-run-deploy.sh. the api on postman is running fine but when I run on Google Chrome I'm getting the error XMLHttpRequest error because of CORS, how can we enable CORS on the server? 

I know I can use `client.example.hello('World')` and it'll work fine, but in my case I have dynamic type in the response so it throws the exception `FormatException: No deserialization found for type dynamic ` and that's why I want to use the below

`
var headers = {
  'Content-Type': 'application/json'
};
var request = http.Request('POST', Uri.parse('http://localhost:8080/example'));
request.body = json.encode({
  "method": "hello",
  "name": "world"
});
request.headers.addAll(headers);

http.StreamedResponse response = await request.send();

if (response.statusCode == 200) {
  print(await response.stream.bytesToString());
}
else {
  print(response.reasonPhrase);
}
`

**Comment by vlidholt** [comment]
You may want to use the webserver if you need to send custom responses from the server. You should be able to set the CORS headers there too. (CORS is already enabled for the generated APIs.)

More info on the webserver here: https://docs.serverpod.dev/concepts/webserver
